### PR TITLE
feat: allow tempDir configuration

### PR DESCRIPTION
**Pull request by Tezkerek**
#73

### CODE REVIEW

File: src/config/pdf.php [hunk 1]
```diff
@@ -7,5 +7,6 @@
 	'subject'               => '',
 	'keywords'              => '',
 	'creator'               => 'Laravel Pdf',
-	'display_mode'          => 'fullpage'
+        'display_mode'          => 'fullpage',
```
Comment (niklasravnsborg) [inline src/config/pdf.php]: Here seems to be a whitespace issue

Comment (niklasravnsborg) [reply src/config/pdf.php]: maybe you can force push fix this?

**Comment by niklasravnsborg** [comment]
Awesome! Thanks for the PR. Will merge this as soon as I find the time! 😌
Merry Christmas 🎅

**Comment by freemanchari** [comment]
I think you should merge this!

**Comment by vpyatnitskiy** [comment]
Yes, please do!

The point is that, when installing a package, composer sets default permissions to all its directories ― which is usually `0755`, so the default `tempDir` is not writable by web server, and so the PDF generation is not working. The permissions are also reset when updating or reinstalling a package, which is somewhat hard to keep track of.

The best I've managed to do so far is to put the `chmod` commands into `post-install-cmd` and `post-update-cmd` of my `composer.json`.
I still don't like the fact that I have a world-writable directory within `vendor/`, which supposedly should contain source code of external packages only ― which is why I, too, vouch for this PR.

I would advise against using `__DIR__ . '../temp/'` as default, though, since Laravel by itself doesn't provide the `temp` directory ― leading to broken install all the same. Apparently MPDF wants a whole directory to itself, so `sys_get_temp_dir()` doesn't work here either.

The best you can do is, probably, `storage_path('app/mpdf')`: Laravel requires the directories under `storage/` to be writable, and `storage/app` always exists. This, of course, still leaves the problem of assigning correct permissions to the `tempDir` ― which is, however, a problem in the `mpdf/mpdf` package itself, and should be fixed there. I've sent a PR there (https://github.com/mpdf/mpdf/pull/624), but its fate is unknown at the moment.

If you don't believe the problem would be fixed by `mpdf/mpdf` team in time, or at all, you're better off with `null` as default. Advanced users would be able to override the default, while regular users would still be able to google the original exception and figure out what to do. Of course, this requires mentioning the new config option in the README file.

What do you think?

**Comment by Tezkerek** [comment]
I think having every user specify their own path would be best, since AFAIK there isn't a default `tmp` location in Laravel.

**Comment by Tezkerek** [comment]
@vpyatnitskiy I see that mpdf/mpdf#624 was merged, does that mean that `storage_path('app/mpdf')` will work now?

**Comment by vpyatnitskiy** [comment]
@Tezkerek As soon as they publish a release with the changes to packagist.org, it will. Right now it appears that the last release was on January 3. So, no, not yet.

**Pull request merged**